databases/week2/Homework/Homework_tables.sql

```diff
@@ -0,0 +1,152 @@
+USE Clinic_scheduling;
+
+CREATE TABLE
+  Clinics (
+    clinic_id INT NOT NULL AUTO_INCREMENT,
+    Clinic_name VARCHAR NOT NULL,
+    Clinic_address VARCHAR NOT NULL,
+    PRIMARY KEY (clinic_id)
+  );
+
+-- a USER can either have work schedule for certain date or a regular working schedule for certain weekdays
+-- hence different kind of time-slots
+CREATE TABLE
+  Working_hours (
+    date DATE,
+    weekday CHAR(3),
+    begins DATETIME NOT NULL,
+    ends DATETIME NOT NULL,
+    slot_id INT NOT NULL AUTO_INCREMENT,
+    PRIMARY KEY (slot_id)
+  );
+
+CREATE TABLE
+  Users (
+    user_id INT NOT NULL AUTO_INCREMENT,
+    name VARCHAR NOT NULL,
+    sex VARCHAR,
+    address VARCHAR,
+    phone INT NOT NULL,
+    email VARCHAR NOT NULL,
+    notes VARCHAR,
+    PRIMARY KEY (user_id)
+  );
+
+CREATE TABLE
+  Medical_forms (
+    Form_id INT NOT NULL AUTO_INCREMENT,
+    Form_name VARCHAR NOT NULL,
+    Form_link VARCHAR NOT NULL,
+    Form_description VARCHAR NOT NULL,
+    PRIMARY KEY (Form_id)
+  );
+
+-- each doctor and several other user roles have specific schedules
+CREATE TABLE
+  Doctors_have_working_hours (
+    schedule_id INT NOT NULL AUTO_INCREMENT,
+    slot_id INT NOT NULL,
+    user_id INT NOT NULL,
+    PRIMARY KEY (schedule_id),
+    FOREIGN KEY (slot_id) REFERENCES Working_hours (slot_id),
+    FOREIGN KEY (user_id) REFERENCES Users (user_id)
+  );
+
+-- the 'main' entity for this database
+CREATE TABLE
+  Visit (
+    cancelled BOOLEAN,
+    Shall_remind BOOLEAN,
+    notes VARCHAR,
+    Diagnosis VARCHAR,
+    visit_id INT NOT NULL AUTO_INCREMENT,
+    schedule_id INT NOT NULL,
+    doc_user_id INT NOT NULL,
+    patient_user_id INT NOT NULL,
+    room_id INT NOT NULL,
+    PRIMARY KEY (visit_id),
+    FOREIGN KEY (schedule_id) REFERENCES Doctors_have_working_hours (schedule_id),
+    FOREIGN KEY (doc_user_id) REFERENCES Users (user_id),
+    FOREIGN KEY (patient_user_id) REFERENCES Users (user_id),
```

Comment:
I would split users into 'patients' and 'doctors', they can and will have different properties and will be more clean implementation

Reply:
I also thoght about having separate tables for patients and doctors. But later I decided they all are users (and beside them there would be several more user-types like nurse, administrator, call-center operator etc.) So, the main logic was to have all users wich essentialy have identical attributes in one table and then refer to the Role of each user. And the Role should potentialy have all the necessary permissons for the specific type of user. E.g. octors can see and edit medical forms and see all patients, patients see neither medical forms, nor other user's data, but can see available slots for visits, call-center doesn't see medical forms, but sees every doctor schedule and every patient's user-info etc.
If separated, then we should probably have a table for each type of user with the same columns. And I am not sure it's a good practical idea (however I don't have much experience to bet my wallet on that :)